### ISSUE TITLE
v-for components reusing socket events

### ISSUE THREAD

**Issue by KodeStar** [probil/vue-socket.io-extended#407]
I have a v-for list that loads in an SFC.
In the SFC i've tried using both `sockets:` and registering a listener with `this.$options.sockets` but neither work properly but in different ways.

In the first case I had:
```js
  sockets: {
    consoleKey (val) { 
      if (val.uuid === this.uuid) this.term.write(val.data.toString('utf8'))
    } 
  },
```
This worked fine until I removed a component (you can add and remove the components at will), as soon as a component was removed you couldn't add any more (you could add them but they no longer respond to the socket event). But before removing a component you could add as many as you wanted and they worked perfectly as expected.

So then I tried
```js
 computed: {
    identifier: function () {
      return 'consoleKey ' + this.uuid
    },
  },
  mounted () {
    this.$options.sockets[this.identifier] = data => {
      this.term.write(data.data.toString('utf8'))
    }
  },
  beforeDestroy () {
    this.term.dispose()
    this.$socket.emit('console_close ' + this.uuid)
    delete this.$options.sockets[this.identifier]
  }
```
With this you can add and remove components fine, however, it seems to register the listener multiple times. So if I add the component 3 times, and type 'ls' in each one, the last one shows correctly as `ls`, the one before it shows as `llss` and the first one shows as `lllsss`. I have tried both ways multiple different ways. Any ideas?

**Comment by KodeStar** [comment]
In the server component I'm doing
```
      .on('data', (data) => {
        socket.emit('consoleKey ' + config.identifier, {
          uuid: config.identifier,
          data: data.toString('utf8')
        });
      })
```
On the `socket:` version I'm just using `socket.emit('consoleKey', {` as I wasn't sure how to register the event with a variable, so I sent it as part of the payload instead, but left it in the payload for the `this.$options.sockets` as well, just because I've changed it so many times

**Comment by KodeStar** [comment]
I wonder if it has anything to do with reactivity?
When I click add a new component, it appends to a data object, that object has a v-for on it. So i'm wondering if the first time I add the object, the v-for loops through the object and adds the listener to item 1, I add a new component, the v-for loops through the items and adds a listener to item 1 and item 2, so now item 1 has 2 listeners on it?

**Comment by probil** [comment]
Hi @KodeStar  :wave: 

Why don't just use Vuex for that? It was designed for data organization in mind. 
In this case you can listen for those events in one place.

BTW which version of the library you are using?

Thanks for your question

**Comment by KodeStar** [comment]
@probil Thanks for your reply. I did start looking at trying to put it in a vuex store, can't remember why I abandoned it. I think it was because if this really was a bug and you took the time to fix it I wanted to be able to confirm the fix worked for you. I'll take another look at vuex though, thanks.

I'm using  vue-socket.io-extended@3.2.2

**Comment by probil** [comment]
@KodeStar Is this still relevant?
- We have found a way to make library work with Quasar
- The first issue you mentioned is the same as in #408 and you can fix it by downgrading to `v3.2.1`
I've found a better way to fix without downgrading. Fix will be available in the upcoming version `v4.x`

**Comment by KodeStar** [comment]
Feel free to close it then :)
